### ISSUE TITLE
[Question] SAC performance on Hopper (mujoco)

### ISSUE THREAD

**Issue by albertcthomas** [DLR-RM/rl-baselines3-zoo#343]
### ❓ Question

Following the discussion in #340 about the performance of SAC on Hopper-mujoco (I should follow the process explained there but in the meantime I am reporting my findings here).

The perf reported in this repo for SAC is 2266.78 +/- 1121.81. I tried running the same SAC (with the same hypers) and I obtained the following  learning curve with 10 evals (showing the mean, min and max) which seems to confirm the reported result.
![Screenshot 2023-01-13 113643](https://user-images.githubusercontent.com/15966638/212299764-9ce7d728-2205-4d79-b97b-9a96ca079cb0.png)

The issue is that it has a very large variance and it is depending a lot on when we stop the training. The results reported in the [original SAC paper](https://arxiv.org/pdf/1801.01290.pdf) seem better and more stable. I can also reach a better and more stable performance with SAC on Hopper (mujoco) than the ones reported in this repo using the hyperparameters of the original SAC paper, i.e. using a reward scaling of 1 instead of using the automatic tuning of the entropy coefficient. Here is the tensorboard logging obtained with sb3 for 4 runs (evaluation is done every 10k steps on 10 episodes).
![image](https://user-images.githubusercontent.com/15966638/212299099-36902d2e-6f64-4c4f-a159-d901e85fb71f.png)

Here you can see the mean, max and min of the evaluation on the 10 episodes
for the four runs:
![image](https://user-images.githubusercontent.com/15966638/212296227-cd13404a-7e2e-4aa6-ab33-167c490f54ef.png)

I'm still investigating this a bit as the performance is still a bit lower than the performance reported in the [original SAC paper](https://arxiv.org/pdf/1801.01290.pdf) where the final perf is closer to 3500.

Finally, [in the paper where they learn the entropy coefficient](https://arxiv.org/pdf/1812.05905.pdf) they seem to have a better (and stable) performance so I am not sure why this is not really working here.

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/rl-baselines3-zoo/issues) in the repo
- [X] I have read the [SB3 documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] I have read the [RL Zoo README](https://github.com/DLR-RM/rl-baselines3-zoo/blob/master/README.md)
- [X] If code there is, it is minimal and working
- [X] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
> Finally, [in the paper where they learn the entropy coefficient](https://arxiv.org/pdf/1812.05905.pdf) they seem to have a better (and stable) performance so I am not sure why this is not really working here.

if you look at Figure 1, the results for SAC with learned temperature are quite noisy too (blue curve) and achieve a max reward around 3000.

You are also comparing different version of MuJoCo (v1 vs v2 vs v3) that used different MuJoCo versions...

**Comment by albertcthomas** [comment]
Thanks the comments.

> if you look at Figure 1, the results for SAC with learned temperature are quite noisy too (blue curve) and achieve a max reward around 3000.

yes you are right, I missed the drop at the end... the max is more around 3500. and it indeed has a larger variance than the orange one (with fixed temperature).

> You are also comparing different versions of MuJoCo envs (v1 vs v2 vs v3) that used different MuJoCo versions...

yes

> EDIT: if you want a more stable learning, maybe just doing linear decay for the learning rate should do the trick (as we do for the PyBullet equivalent)

I actually tried this without success, I guess I will have to tweak the initial learning rate a bit more to see if I can reach reasonable performance

**Comment by araffin** [comment]
Closing in favor of https://github.com/openrlbenchmark/openrlbenchmark/issues/7

The runs are here: https://wandb.ai/openrlbenchmark/sb3?workspace=user-araffin
